includes/multi-currency/Analytics.php

```diff
@@ -320,7 +341,7 @@ public function filter_where_clauses( array $clauses ): array {
 		}
 
 		$prefix       = 'wcpay_multicurrency_';
-		$currency_tbl = $prefix . 'currency_postmeta';
+		$currency_tbl = $prefix . 'currency_meta';
```

Comment:
One small thing, we need to also do the check to see which table/column to use for the WHERE clause when filtering by currency. I tested this code locally and it will achieve this (replaces lines 344-355). 
```suggestion
		$prefix       = 'wcpay_multicurrency_';
		$currency_tbl = $this->is_cot_enabled()
			? $prefix . 'order_currency'
			: $prefix . 'currency_meta';

		$currency_args = $this->get_customer_currency_args_from_request();
		if ( ! empty( $currency_args['currency_is'] ) ) {
			$currency_is = sprintf( "'%s'", implode( "', '", $currency_args['currency_is'] ) );
			if ( $this->is_cot_enabled() ) {
				$clauses[] = "AND {$currency_tbl}.currency IN ({$currency_is})";
			} else {
				$clauses[] = "AND {$currency_tbl}.meta_value IN ({$currency_is})";
			}

		}

		if ( ! empty( $currency_args['currency_is_not'] ) ) {
			$currency_is_not = sprintf( "'%s'", implode( "', '", $currency_args['currency_is_not'] ) );
			if ( $this->is_cot_enabled() ) {
				$clauses[] = "AND {$currency_tbl}.currency NOT IN ({$currency_is_not})";
			} else {
				$clauses[] = "AND {$currency_tbl}.meta_value NOT IN ({$currency_is_not})";
			}
		}
```

Reply:
Another great catch!

I implemented this change in https://github.com/Automattic/woocommerce-payments/pull/4677/commits/203339e06589c79cb61fed3afe708b02900839c7, but by pre-loading the field instead of adding `if-else` statements.